src/compas/utilities/itertools_.py

```diff
@@ -48,7 +48,8 @@
     'first_true',
     'random_permutation',
     'random_combination',
-    'random_combination_with_replacement'
+    'random_combination_with_replacement',
+    'match_length'
```

Comment:
perhaps a good name would be something more in line with how a similar function is named in `numpy` (https://docs.scipy.org/doc/numpy/reference/generated/numpy.pad.html). so something like `to_padded_list`?

in rhino there are a lot of `coerce` functions. perhaps `coerce_list`?

in `numpy` there is also something like `array_like` and `empty_like` => `list_like`?

Reply:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.empty_like.html